File: frontend/src/types/Study.ts
export interface SourceExcerpt {
  document_id: number
  document_filename: string
  page_number: number | null
  similarity: number
  excerpt: string
}

export interface StudyChatResponse {
  question: string
  answer: string
  model: string
  sources: SourceExcerpt[]
  prompt: string
}

File: backend/app/schemas/study.py
from pydantic import BaseModel


class StudyChatRequest(BaseModel):
    question: str
    course_id: int
    top_k: int = 4


class SourceExcerpt(BaseModel):
    document_id: int
    document_filename: str
    page_number: int | None
    similarity: float
    excerpt: str


class StudyChatResponse(BaseModel):
    question: str
    answer: str
    model: str
    sources: list[SourceExcerpt]
    prompt: str
